network_wrangler/RoadwayNetwork.py

```diff
@@ -118,41 +118,57 @@ def write(self, filename: str, path: str = '.') -> None:
         shapes_file = os.path.join(path, filename + "_shape.geojson")
         self.shapes_df.to_file(shapes_file, driver='GeoJSON')
 
-
-
-    def apply_roadway_feauture_change(self, project_card: str) -> bool:
+    def apply_roadway_feauture_change(self, card_dict: dict) -> bool:
         '''
-        Changes the road network according to the project card passed
-        
+        Changes the road network according to the project card information passed
+
         args:
-        project_card: the path to the project card
-        
+        card_dict: dictionary with project card information
+
         returns:
         bool: True if successful.
         '''
-        #import pdb; pdb.set_trace()
-        with open(project_card, 'r') as card:
-            card_dict = yaml.safe_load(card)
-            
-            road_dict = card_dict.get("Road")            
-            name_str = road_dict.get(list(road_dict)[0])
-            id = name_str.split('=')[1]
-            
-            change_dict = card_dict.get("Change")
-            existing_value = change_dict.get(list(change_dict)[0])
-            build_value = change_dict.get(list(change_dict)[1])
-            
-            attribute = card_dict.get("Attribute")
-            
-            print(id)
-            print(attribute)
-            print(existing_value)
-            print(build_value)
-            
-            
-            #check existing value that it matches the attribute field in self.links_df
-            #TODO: how do i manipulate the data in the dataframe to apply the changes?
-            
-            self.links_df
-            
+
+        road_dict = card_dict.get("Road")
+        road_id = road_dict.get("Name").split("=")[1]
+
+        attribute = card_dict.get("Attribute").upper()
+        change_dict = card_dict.get("Change")
+        existing_value = change_dict.get("Existing")
+        build_value = change_dict.get("Build")
+
+        # identify the network link with same id as project card road id
+        # check if the attribute to be updated exists on the network links
+        # get the current attribute value for the link from the network
+        # check for current attribute value to be same as defined in the project card
+        # update the link attribute value
+
+        # TODO:
+        # change desired logger level for the different checks
+
+        found = False
+        for link in self.links_df['features']:
+            if link['id'] == road_id:
```

Comment:
this selection matching should be based on the ID type that is used, it could be shared streets, osm, model id, etc.